### DISCUSSION TITLE
Lesson 6: Forking Networks not happening (?)

### DISCUSSION THREAD

**Discussion by tsopanid** [smartcontractkit/full-blockchain-solidity-course-py#62] [Q&A] [answered]
This happens with both Infura and Alchemy

I add the network:
`brownie networks add development mainnet-fork-dev cmd=ganache-cli host=http://172.0.0.1 fork=https://eth-mainnet.alchemyapi.io/v2/<REMOVED_PROJECT_ID> accounts=10 mnemonic=brownie port=8545`

but then when i try to deploy using:  
`brownie run scripts/deploy.py --network mainnet-fork-dev`
my terminal hangs with this output:
```
Brownie v1.16.3 - Python development framework for Ethereum

BrownieFundMeProject is the active project.

Launching 'ganache-cli --accounts 10 --fork https://mainnet.infura.io/v3/<REMOVED_PROJECT_ID> --mnemonic brownie --port 8545 --hardfork istanbul'...
```

i waited for half an hour, but nothing happened.

the same happens when i try to deploy afeter i remove the network and use infura:  
`brownie networks add development mainnet-fork-dev cmd=ganache-cli host=http://172.0.0.1 fork='https://mainnet.infura.io/v3/$WEB3_INFURA_PROJECT_ID' accounts=10 mnemonic=brownie port=8545`

<REMOVED_PROJECT_ID> of course is replaced by my real one when i run the command.  

Any ideas?

**Comment by PatrickAlphaC** [comment, marked as answer]
The issue is here:
```
Launching 'ganache-cli --accounts 10 --fork https://mainnet.infura.io/v3/<REMOVED_PROJECT_ID> --mnemonic brownie --port 8545 --hardfork istanbul
```

For some reason, it's assigning infura to be your forking URL, when it should be assigning alchemy. What happens when you output `brownie networks list True`?

(Please don't upload your projectIds here as you've been doing. Thank you!)

**Comment by tsopanid** [comment]
Sorry i mixed things up here while posting, because i tried both infura and alchemy.
The output from above is from the infura test.
For alchemy i get:
`Launching 'ganache-cli --accounts 10 --fork https://eth-mainnet.alchemyapi.io/v2/<REMOVED_PROJECT_ID> --mnemonic brownie --port 8545 --hardfork istanbul'...
`
The output of `brownie list True` is:
```
Ethereum
  ├─Mainnet (Infura)
  │ ├─id: mainnet
  │ ├─chainid: 1
  │ ├─explorer: https://api.etherscan.io/api
  │ ├─host: https://mainnet.infura.io/v3/$WEB3_INFURA_PROJECT_ID
  │ └─multicall2: 0x5BA1e12693Dc8F9c48aAD8770482f4739bEeD696
  ├─Ropsten (Infura)
  │ ├─id: ropsten
  │ ├─chainid: 3
  │ ├─explorer: https://api-ropsten.etherscan.io/api
  │ ├─host: https://ropsten.infura.io/v3/$WEB3_INFURA_PROJECT_ID
  │ └─multicall2: 0x5BA1e12693Dc8F9c48aAD8770482f4739bEeD696
  ├─Rinkeby (Infura)
  │ ├─id: rinkeby
  │ ├─chainid: 4
  │ ├─explorer: https://api-rinkeby.etherscan.io/api
  │ ├─host: https://rinkeby.infura.io/v3/$WEB3_INFURA_PROJECT_ID
  │ └─multicall2: 0x5BA1e12693Dc8F9c48aAD8770482f4739bEeD696
  ├─Goerli (Infura)
  │ ├─id: goerli
  │ ├─chainid: 5
  │ ├─explorer: https://api-goerli.etherscan.io/api
  │ ├─host: https://goerli.infura.io/v3/$WEB3_INFURA_PROJECT_ID
  │ └─multicall2: 0x5BA1e12693Dc8F9c48aAD8770482f4739bEeD696
  ├─Kovan (Infura)
  │ ├─id: kovan
  │ ├─chainid: 42
  │ ├─explorer: https://api-kovan.etherscan.io/api
  │ ├─host: https://kovan.infura.io/v3/$WEB3_INFURA_PROJECT_ID
  │ └─multicall2: 0x5BA1e12693Dc8F9c48aAD8770482f4739bEeD696
  └─ganache-local
    ├─id: ganache-local
    ├─chainid: 1337
    └─host: http://127.0.0.1:8545

Ethereum Classic
  ├─Mainnet
  │ ├─id: etc
  │ ├─chainid: 61
  │ ├─explorer: https://blockscout.com/etc/mainnet/api
  │ └─host: https://www.ethercluster.com/etc
  └─Kotti
    ├─id: kotti
    ├─chainid: 6
    ├─explorer: https://blockscout.com/etc/kotti/api
    └─host: https://www.ethercluster.com/kotti

Arbitrum
  └─Mainnet
    ├─id: arbitrum-main
    ├─chainid: 42161
    ├─explorer: https://api.arbiscan.io/api
    └─host: https://arb1.arbitrum.io/rpc

Binance Smart Chain
  ├─Testnet
  │ ├─id: bsc-test
  │ ├─chainid: 97
  │ ├─explorer: https://api-testnet.bscscan.com/api
  │ └─host: https://data-seed-prebsc-1-s1.binance.org:8545
  └─Mainnet
    ├─id: bsc-main
    ├─chainid: 56
    ├─explorer: https://api.bscscan.com/api
    └─host: https://bsc-dataseed.binance.org

Fantom Opera
  ├─Testnet
  │ ├─id: ftm-test
  │ ├─chainid: 4002
  │ ├─explorer: https://explorer.testnet.fantom.network
  │ └─host: https://rpc.testnet.fantom.network
  └─Mainnet
    ├─id: ftm-main
    ├─chainid: 250
    ├─explorer: https://api.ftmscan.com/api
    └─host: https://rpcapi.fantom.network

Polygon
  ├─Mainnet (Infura)
  │ ├─id: polygon-main
  │ ├─chainid: 137
  │ ├─explorer: https://api.polygonscan.com/api
  │ ├─host: https://polygon-mainnet.infura.io/v3/$WEB3_INFURA_PROJECT_ID
  │ └─multicall2: 0xc8E51042792d7405184DfCa245F2d27B94D013b6
  └─Mumbai Testnet (Infura)
    ├─id: polygon-test
    ├─chainid: 80001
    ├─explorer: https://api-testnet.polygonscan.com/api
    ├─host: https://polygon-mumbai.infura.io/v3/$WEB3_INFURA_PROJECT_ID
    └─multicall2: 0x6842E0412AC1c00464dc48961330156a07268d14

XDai
  ├─Mainnet
  │ ├─id: xdai-main
  │ ├─chainid: 100
  │ ├─explorer: https://blockscout.com/xdai/mainnet/api
  │ └─host: https://xdai.poanetwork.dev
  └─Testnet
    ├─id: xdai-test
    ├─chainid: 77
    ├─explorer: https://blockscout.com/poa/sokol/api
    └─host: https://sokol.poa.network

Development
  ├─Ganache-CLI
  │ ├─id: development
  │ ├─cmd: ganache-cli
  │ └─host: http://127.0.0.1
  │   ├─accounts: 10
  │   ├─evm_version: istanbul
  │   ├─gas_limit: 12000000
  │   ├─mnemonic: brownie
  │   └─port: 8545
  ├─Geth Dev
  │ ├─id: geth-dev
  │ ├─cmd: ethnode
  │ └─host: http://127.0.0.1
  │   └─port: 8545
  ├─Hardhat
  │ ├─id: hardhat
  │ ├─cmd: npx hardhat node
  │ └─host: http://127.0.0.1
  │   └─port: 8545
  ├─Hardhat (Mainnet Fork)
  │ ├─id: hardhat-fork
  │ ├─cmd: npx hardhat node
  │ ├─host: http://127.0.0.1
  │ └─timeout: 120
  │   ├─fork: mainnet
  │   └─port: 8545
  ├─Ganache-CLI (Mainnet Fork)
  │ ├─id: mainnet-fork
  │ ├─cmd: ganache-cli
  │ ├─host: http://127.0.0.1
  │ └─timeout: 120
  │   ├─accounts: 10
  │   ├─evm_version: istanbul
  │   ├─fork: mainnet
  │   ├─gas_limit: 12000000
  │   ├─mnemonic: brownie
  │   └─port: 8545
  ├─Ganache-CLI (BSC-Mainnet Fork)
  │ ├─id: bsc-main-fork
  │ ├─cmd: ganache-cli
  │ ├─host: http://127.0.0.1
  │ └─timeout: 120
  │   ├─accounts: 10
  │   ├─evm_version: istanbul
  │   ├─fork: bsc-main
  │   ├─gas_limit: 12000000
  │   ├─mnemonic: brownie
  │   └─port: 8545
  ├─Ganache-CLI (FTM-Mainnet Fork)
  │ ├─id: ftm-main-fork
  │ ├─cmd: ganache-cli
  │ ├─host: http://127.0.0.1
  │ └─timeout: 120
  │   ├─accounts: 10
  │   ├─evm_version: istanbul
  │   ├─fork: ftm-main
  │   ├─gas_limit: 12000000
  │   ├─mnemonic: brownie
  │   └─port: 8545
  ├─Ganache-CLI (Polygon-Mainnet Fork)
  │ ├─id: polygon-main-fork
  │ ├─cmd: ganache-cli
  │ ├─host: http://127.0.0.1
  │ └─timeout: 120
  │   ├─accounts: 10
  │   ├─evm_version: istanbul
  │   ├─fork: polygon-main
  │   ├─gas_limit: 20000000
  │   ├─mnemonic: brownie
  │   └─port: 8545
  ├─Ganache-CLI (XDai-Mainnet Fork)
  │ ├─id: xdai-main-fork
  │ ├─cmd: ganache-cli
  │ ├─host: http://127.0.0.1
  │ └─timeout: 120
  │   ├─accounts: 10
  │   ├─evm_version: istanbul
  │   ├─fork: xdai-main
  │   ├─gas_limit: 20000000
  │   ├─mnemonic: brownie
  │   └─port: 8545
  └─mainnet-fork-dev
    ├─id: mainnet-fork-dev
    ├─cmd: ganache-cli
    └─host: http://172.0.0.1
      ├─accounts: 10
      ├─fork: https://eth-mainnet.alchemyapi.io/v2/<REMOVED_PROJECT_ID>
      ├─mnemonic: brownie
      └─port: 854
```

**Reply by tsopanid** [reply]
Now i waited for a longer period and finally got some errors:
```
brownie run scripts/deploy.py --network mainnet-fork-dev                                                                                                                          21-09-21 - 15:28:07Brownie v1.16.3 - Python development framework for Ethereum

BrownieFundMeProject is the active project.

Launching 'ganache-cli --accounts 10 --fork https://eth-mainnet.alchemyapi.io/v2/<REMOVED> --mnemonic brownie --port 8545 --hardfork istanbul'...
Terminating local RPC client...
  File "brownie/_cli/__main__.py", line 64, in main
    importlib.import_module(f"brownie._cli.{cmd}").main()
  File "brownie/_cli/run.py", line 43, in main
    network.connect(CONFIG.argv["network"])
  File "brownie/network/main.py", line 50, in connect
    rpc.launch(active["cmd"], **active["cmd_settings"])
  File "brownie/network/rpc/__init__.py", line 95, in launch
    raise RPCConnectionError(cmd, self.process, uri)
  File "brownie/exceptions.py", line 41, in __init__
    f"\n\nCommand: {cmd}\nURI: {uri}\nExit Code: {proc.poll()}"
AttributeError: 'NoneType' object has no attribute 'poll'
```

if i run   
`ganache-cli --accounts 10 --fork https://eth-mainnet.alchemyapi.io/v2/<REMOVED> --mnemonic brownie --port 8545 --hardfork istanbul`  
ganache does load:
```
Ganache CLI v6.12.2 (ganache-core: 2.13.2)

Available Accounts
==================
(0) ... (100 ETH)
...
(9) ... (100 ETH)

Private Keys
==================
(0) ...
...
(9) ...

HD Wallet
==================
Mnemonic:      brownie
Base HD Path:  m/44'/60'/0'/0/{account_index}

Gas Price
==================
20000000000

Gas Limit
==================
6721975

Call Gas Limit
==================
9007199254740991

Forked Chain
==================
Location:       https://eth-mainnet.alchemyapi.io/v2/<REMOVED>
Block:          13269548
Network ID:     1
Time:           Tue Sep 21 2021 16:59:05 GMT+0200 (Mitteleuropäische Sommerzeit)
Max Cache Size: 1073741824 bytes

Listening on 127.0.0.1:8545
```
(Available Accounts and Private Keys removed ...)

I also cloned https://github.com/PatrickAlphaC/brownie_fund_me.git, same happening there...

**Reply by tsopanid** [reply]
No idea what was going on.
The Problem is now solved.
After continuing with the course, the test for the Smart Contract Lottery worked. 
So i deleted and added mainnet-fork-dev again.
No it is working!

Thank you for your halp and also for the really excelent course!

**Reply by PatrickAlphaC** [reply]
Welp.... if you ever come across what got it to work, and want to come back and update this, let me know!

Great work!

**Reply by harishvishwakarma** [reply]
Stuck with the same issue

```
brownie run scripts/deploy.py --network mainnet-forkBrownie v1.17.0 - Python development framework for Ethereum

SmartcontractLotteryProject is the active project.

Launching 'ganache-cli --accounts 10 --mnemonic brownie --port 8545 --hardfork istanbul'...
Terminating local RPC client...
  File "brownie/_cli/__main__.py", line 64, in main
    importlib.import_module(f"brownie._cli.{cmd}").main()
  File "brownie/_cli/run.py", line 44, in main
    network.connect(CONFIG.argv["network"])
  File "brownie/network/main.py", line 50, in connect
    rpc.launch(active["cmd"], **active["cmd_settings"])
  File "brownie/network/rpc/__init__.py", line 95, in launch
    raise RPCConnectionError(cmd, self.process, uri)
  File "brownie/exceptions.py", line 41, in __init__
    f"\n\nCommand: {cmd}\nURI: {uri}\nExit Code: {proc.poll()}"
AttributeError: 'NoneType' object has no attribute 'poll'
```

**Reply by harishvishwakarma** [reply]
@PatrickAlphaC Do you know the solution to this one?

**Reply by harishvishwakarma** [reply]
Ahh! I know what went wrong. Here is the solution for others like me.

The problem is in the command that I ran for creating the 'mainnet-fork' network.

```brownie networks add development mainnet-fork cmd=ganache-cli host=https://eth-mainnet.alchemyapi.io/XXX accounts=10 mnemonic=brownie port=8545```

This is using wrong host and is missing the fork. The command should rather have been

```brownie networks add development mainnet-fork cmd=ganache-cli host=http://127.0.0.1 fork=https://eth-mainnet.alchemyapi.io/XXX accounts=10 mnemonic=brownie port=8545```

**Reply by PatrickAlphaC** [reply]
nice work!

**Comment by mohzourob** [comment]
Hi, 
I have add the network used this command
`brownie networks add development mainnet-fork-dev cmd=ganache-cli host=http://127.0.0.1 fork=https://eth-mainnet.alchemyapi.io/v2/xxxx accounts=10 mnemonic=brownie port=8545 `

when I run `brownie run scripts/deploy.py --network mainnet-fork-dev ` Not working, pedding and when I cancel the request give me this error

^CTraceback (most recent call last):
  File "/home/dashzou/.local/bin/brownie", line 8, in <module>
    sys.exit(main())
  File "/home/dashzou/.local/pipx/venvs/eth-brownie/lib/python3.9/site-packages/brownie/_cli/__main__.py", line 64, in main
    importlib.import_module(f"brownie._cli.{cmd}").main()
  File "/home/dashzou/.local/pipx/venvs/eth-brownie/lib/python3.9/site-packages/brownie/_cli/run.py", line 44, in main
    network.connect(CONFIG.argv["network"])
  File "/home/dashzou/.local/pipx/venvs/eth-brownie/lib/python3.9/site-packages/brownie/network/main.py", line 50, in connect
    rpc.launch(active["cmd"], **active["cmd_settings"])
  File "/home/dashzou/.local/pipx/venvs/eth-brownie/lib/python3.9/site-packages/brownie/network/rpc/__init__.py", line 83, in launch
    if web3.isConnected():
  File "/home/dashzou/.local/pipx/venvs/eth-brownie/lib/python3.9/site-packages/brownie/network/web3.py", line 115, in isConnected
    return super().isConnected()
  File "/home/dashzou/.local/pipx/venvs/eth-brownie/lib/python3.9/site-packages/web3/main.py", line 327, in isConnected
    return self.provider.isConnected()
  File "/home/dashzou/.local/pipx/venvs/eth-brownie/lib/python3.9/site-packages/web3/providers/base.py", line 103, in isConnected
    response = self.make_request(RPCEndpoint('web3_clientVersion'), [])
  File "/home/dashzou/.local/pipx/venvs/eth-brownie/lib/python3.9/site-packages/web3/providers/rpc.py", line 88, in make_request
    raw_response = make_post_request(
  File "/home/dashzou/.local/pipx/venvs/eth-brownie/lib/python3.9/site-packages/web3/_utils/request.py", line 48, in make_post_request
    response = session.post(endpoint_uri, data=data, *args, **kwargs)  # type: ignore
  File "/home/dashzou/.local/pipx/venvs/eth-brownie/lib/python3.9/site-packages/requests/sessions.py", line 590, in post
    return self.request('POST', url, data=data, json=json, **kwargs)
  File "/home/dashzou/.local/pipx/venvs/eth-brownie/lib/python3.9/site-packages/requests/sessions.py", line 542, in request
    resp = self.send(prep, **send_kwargs)
  File "/home/dashzou/.local/pipx/venvs/eth-brownie/lib/python3.9/site-packages/requests/sessions.py", line 655, in send
    r = adapter.send(request, **kwargs)
  File "/home/dashzou/.local/pipx/venvs/eth-brownie/lib/python3.9/site-packages/requests/adapters.py", line 439, in send
    resp = conn.urlopen(
  File "/home/dashzou/.local/pipx/venvs/eth-brownie/lib/python3.9/site-packages/urllib3/connectionpool.py", line 699, in urlopen
    httplib_response = self._make_request(
  File "/home/dashzou/.local/pipx/venvs/eth-brownie/lib/python3.9/site-packages/urllib3/connectionpool.py", line 445, in _make_request
    six.raise_from(e, None)
  File "<string>", line 3, in raise_from
  File "/home/dashzou/.local/pipx/venvs/eth-brownie/lib/python3.9/site-packages/urllib3/connectionpool.py", line 440, in _make_request
    httplib_response = conn.getresponse()
  File "/usr/lib/python3.9/http/client.py", line 1371, in getresponse
    response.begin()
  File "/usr/lib/python3.9/http/client.py", line 319, in begin
    version, status, reason = self._read_status()
  File "/usr/lib/python3.9/http/client.py", line 280, in _read_status
    line = str(self.fp.readline(_MAXLINE + 1), "iso-8859-1")
  File "/usr/lib/python3.9/socket.py", line 704, in readinto
    return self._sock.recv_into

**Reply by mohzourob** [reply]
@PatrickAlphaC 
Could u help me with this issue pls ?

**Reply by mohzourob** [reply]
@PatrickAlphaC fixed, when I used socket wss://eth-mainnet.alchemyapi.io/v2/xx worked fine,

**Reply by PatrickAlphaC** [reply]
perfect!

**Reply by stephenthumb** [reply]
Confirmed dash7ou's solution.  The Alchemy websockets api url works fine in Brownie just replacing the http url directly in the fork value for the network definitiion.  I tested the HTTP URL endpoint with Web3 directly and was able to call ETH block numbers and gas prices so unsure what Brownie is doing specifically that doesn't work.

brownie networks add development mainnet-fork-dev cmd=ganache-cli host=http://127.0.0.1 fork='wss://eth-mainnet.alchemyapi.io/v2/xxxxxxxxxxxxxx' accounts=10 mnemonic=brownie port=8545

**Reply by stephenthumb** [reply]
Also found after getting the successful connection using the websockets url the http endpoint started working also.

**Comment by gorkemkeles** [comment]
Hi all, I have the same error while running (AttributeError: 'NoneType' object has no attribute 'poll')

`brownie run scripts/deploy.py --network mainnet-fork-dev` 

Also, I'm using Alchemy btw.
May I ask how can I remove a **mainnet-fork-dev** from brownie network list then set it again? Because when I set **"port=7545"**  instead of "8545" because it was 7545 in my Ganache!

**Reply by cromewar** [reply]
Hi @gorkemkeles deleting a network is pretty easy, I've created a new network called "testone"

![image](https://user-images.githubusercontent.com/20199180/139535689-571b46ab-7809-4a03-8b60-89471fb128ff.png)

For delete it I must type:

`brownie networks delete testone` (mainnet-fork-dev in your case) and as you can see the network is not there anymore.

![image](https://user-images.githubusercontent.com/20199180/139535792-117efacc-35dc-429f-bc3b-56e8a56755e5.png)

**Reply by gorkemkeles** [reply]
@cromewar  thank you so much

**Reply by PatrickAlphaC** [reply]
@cromewar a beast :)

**Comment by Mach0p** [comment]
Hi, did anyone find the solution to this? I'm running into the same problem as it seems it's not forking correctly, and therefor not launching the mainnet-fork-dev network (Timestamp from the video 6:01:45)
@PatrickAlphaC @tsopanid 
```
ganache --version
ganache v7.0.3 (@ganache/cli: 0.1.4, @ganache/core: 0.1.4)

brownie --version
INFO: Could not find files for the given pattern(s).
Brownie v1.16.4 - Python development framework for Ethereum

python --version
Python 3.10.4
```

I added mainnet-fork-dev with `brownie networks add development mainnet-fork-dev cmd=ganache-cli host=https://127.0.0.1 fork='https://eth-mainnet.alchemyapi.io/v2/$MY_KEY$' accounts=10 mnemonic=brownie port=8545`
(I tried the fork url with ', " and same result.)

When running networks list true i get :
```
Ethereum
  ├─Mainnet (Infura)
  │ ├─id: mainnet
  │ ├─chainid: 1
  │ ├─explorer: https://api.etherscan.io/api
  │ ├─host: https://mainnet.infura.io/v3/$WEB3_INFURA_PROJECT_ID
  │ └─multicall2: 0x5BA1e12693Dc8F9c48aAD8770482f4739bEeD696
  ├─Ropsten (Infura)
  │ ├─id: ropsten
  │ ├─chainid: 3
  │ ├─explorer: https://api-ropsten.etherscan.io/api
  │ ├─host: https://ropsten.infura.io/v3/$WEB3_INFURA_PROJECT_ID
  │ └─multicall2: 0x5BA1e12693Dc8F9c48aAD8770482f4739bEeD696
  ├─Rinkeby (Infura)
  │ ├─id: rinkeby
  │ ├─chainid: 4
  │ ├─explorer: https://api-rinkeby.etherscan.io/api
  │ ├─host: https://rinkeby.infura.io/v3/$WEB3_INFURA_PROJECT_ID
  │ └─multicall2: 0x5BA1e12693Dc8F9c48aAD8770482f4739bEeD696
  ├─Goerli (Infura)
  │ ├─id: goerli
  │ ├─chainid: 5
  │ ├─explorer: https://api-goerli.etherscan.io/api
  │ ├─host: https://goerli.infura.io/v3/$WEB3_INFURA_PROJECT_ID
  │ └─multicall2: 0x5BA1e12693Dc8F9c48aAD8770482f4739bEeD696
  ├─Kovan (Infura)
  │ ├─id: kovan
  │ ├─chainid: 42
  │ ├─explorer: https://api-kovan.etherscan.io/api
  │ ├─host: https://kovan.infura.io/v3/$WEB3_INFURA_PROJECT_ID
  │ └─multicall2: 0x5BA1e12693Dc8F9c48aAD8770482f4739bEeD696
  └─ganache-local
    ├─id: ganache-local
    ├─chainid: 5777
    └─host: http://127.0.0.1:8545

Ethereum Classic
  ├─Mainnet
  │ ├─id: etc
  │ ├─chainid: 61
  │ ├─explorer: https://blockscout.com/etc/mainnet/api
  │ └─host: https://www.ethercluster.com/etc
  └─Kotti
    ├─id: kotti
    ├─chainid: 6
    ├─explorer: https://blockscout.com/etc/kotti/api
    └─host: https://www.ethercluster.com/kotti

Arbitrum
  └─Mainnet
    ├─id: arbitrum-main
    ├─chainid: 42161
    ├─explorer: https://api.arbiscan.io/api
    ├─host: https://arb1.arbitrum.io/rpc
    └─multicall2: 0x5B5CFE992AdAC0C9D48E05854B2d91C73a003858

Binance Smart Chain
  ├─Testnet
  │ ├─id: bsc-test
  │ ├─chainid: 97
  │ ├─explorer: https://api-testnet.bscscan.com/api
  │ └─host: https://data-seed-prebsc-1-s1.binance.org:8545
  └─Mainnet
    ├─id: bsc-main
    ├─chainid: 56
    ├─explorer: https://api.bscscan.com/api
    └─host: https://bsc-dataseed.binance.org

Fantom Opera
  ├─Testnet
  │ ├─id: ftm-test
  │ ├─chainid: 4002
  │ ├─explorer: https://explorer.testnet.fantom.network
  │ └─host: https://rpc.testnet.fantom.network
  └─Mainnet
    ├─id: ftm-main
    ├─chainid: 250
    ├─explorer: https://api.ftmscan.com/api
    └─host: https://rpcapi.fantom.network

Polygon
  ├─Mainnet (Infura)
  │ ├─id: polygon-main
  │ ├─chainid: 137
  │ ├─explorer: https://api.polygonscan.com/api
  │ ├─host: https://polygon-mainnet.infura.io/v3/$WEB3_INFURA_PROJECT_ID
  │ └─multicall2: 0xc8E51042792d7405184DfCa245F2d27B94D013b6
  └─Mumbai Testnet (Infura)
    ├─id: polygon-test
    ├─chainid: 80001
    ├─explorer: https://api-testnet.polygonscan.com/api
    ├─host: https://polygon-mumbai.infura.io/v3/$WEB3_INFURA_PROJECT_ID
    └─multicall2: 0x6842E0412AC1c00464dc48961330156a07268d14

XDai
  ├─Mainnet
  │ ├─id: xdai-main
  │ ├─chainid: 100
  │ ├─explorer: https://blockscout.com/xdai/mainnet/api
  │ └─host: https://xdai.poanetwork.dev
  └─Testnet
    ├─id: xdai-test
    ├─chainid: 77
    ├─explorer: https://blockscout.com/poa/sokol/api
    └─host: https://sokol.poa.network

Development
  ├─Ganache-CLI
  │ ├─id: development
  │ ├─cmd: ganache-cli
  │ └─host: http://127.0.0.1
  │   ├─accounts: 10
  │   ├─evm_version: istanbul
  │   ├─gas_limit: 12000000
  │   ├─mnemonic: brownie
  │   └─port: 8545
  ├─Geth Dev
  │ ├─id: geth-dev
  │ ├─cmd: ethnode
  │ └─host: http://127.0.0.1
  │   └─port: 8545
  ├─Hardhat
  │ ├─id: hardhat
  │ ├─cmd: npx hardhat node
  │ └─host: http://127.0.0.1
  │   └─port: 8545
  ├─Hardhat (Mainnet Fork)
  │ ├─id: hardhat-fork
  │ ├─cmd: npx hardhat node
  │ ├─host: http://127.0.0.1
  │ └─timeout: 120
  │   ├─fork: mainnet
  │   └─port: 8545
  ├─Ganache-CLI (BSC-Mainnet Fork)
  │ ├─id: bsc-main-fork
  │ ├─cmd: ganache-cli
  │ ├─host: http://127.0.0.1
  │ └─timeout: 120
  │   ├─accounts: 10
  │   ├─evm_version: istanbul
  │   ├─fork: bsc-main
  │   ├─gas_limit: 12000000
  │   ├─mnemonic: brownie
  │   └─port: 8545
  ├─Ganache-CLI (FTM-Mainnet Fork)
  │ ├─id: ftm-main-fork
  │ ├─cmd: ganache-cli
  │ ├─host: http://127.0.0.1
  │ └─timeout: 120
  │   ├─accounts: 10
  │   ├─evm_version: istanbul
  │   ├─fork: ftm-main
  │   ├─gas_limit: 12000000
  │   ├─mnemonic: brownie
  │   └─port: 8545
  ├─Ganache-CLI (Polygon-Mainnet Fork)
  │ ├─id: polygon-main-fork
  │ ├─cmd: ganache-cli
  │ ├─host: http://127.0.0.1
  │ └─timeout: 120
  │   ├─accounts: 10
  │   ├─evm_version: istanbul
  │   ├─fork: polygon-main
  │   ├─gas_limit: 20000000
  │   ├─mnemonic: brownie
  │   └─port: 8545
  ├─Ganache-CLI (XDai-Mainnet Fork)
  │ ├─id: xdai-main-fork
  │ ├─cmd: ganache-cli
  │ ├─host: http://127.0.0.1
  │ └─timeout: 120
  │   ├─accounts: 10
  │   ├─evm_version: istanbul
  │   ├─fork: xdai-main
  │   ├─gas_limit: 20000000
  │   ├─mnemonic: brownie
  │   └─port: 8545
  ├─mainnet-fork
  │ ├─id: mainnet-fork
  │ ├─cmd: ganache-cli
  │ └─host: http://127.0.0.1
  │   ├─accounts: 10
  │   ├─mnemonic: brownie
  │   └─port: 8545
  └─mainnet-fork-dev
    ├─id: mainnet-fork-dev
    ├─cmd: ganache-cli
    └─host: https://127.0.0.1
      ├─accounts: 10
      ├─fork: https://eth-mainnet.alchemyapi.io/v2/$MY_KEY$
      ├─mnemonic: brownie
      └─port: 8545
```
Then by running : `brownie console --network mainnet-fork-dev` I get: 

```
INFO: Could not find files for the given pattern(s).
Brownie v1.16.4 - Python development framework for Ethereum

BrownieFundMeProject is the active project.

Launching 'ganache-cli.cmd --accounts 10 --fork https://eth-mainnet.alchemyapi.io/v2/$MY_KEY$ --mnemonic brownie --port 8545 --hardfork istanbul'...
```

And it just stays there...I tried with the WSS http address and the same happens.  By also running deploy.py it shows the same as above.
Looking at Alchemy dashboard I can see requests coming in.

**Reply by cromewar** [reply]
as you are trying to use brownie console and your key format is $My_Key$ be sure to export the env variable to path on windows before running this.
Be sure to have closes ganache ui before oppening this, and also is preferable to use python version 3.8 or 3.9 instead of 3.10.

**Reply by Mach0p** [reply]
Hi crome war, thks for your help!
I use the real key, in the url $my_key$ was to not add it publicly here. Ganache UI is closed. And i just went though installing python 3.9.5 and the networks keeps on launching, but there s a slight difference in the message:
Launching '**ganache-cli.cmd --chain.vmErrorsOnRPCResponse** true --wallet.totalAccounts 10 --fork.url https://eth-mainnet.alchemyapi.io/v2/$MY_KEY$ --wallet.mnemonic brownie --server.port 8545 --hardfork istanbul'...

**Reply by cromewar** [reply]
Ok you made a step on the correct direction, the error is now complaining for the rpc client, does ganache-cli runs properly by it's own?
Additionally I strongly suggest you to go to your home folder path, and delete the `.brownie` folder this will erase all the configs and you may add the networks again. This may help.

**Comment by formulaRoot** [comment]
Running into an issue where its asking for the Explorer API

Value Error: Explorer API not set for this network

**Reply by cromewar** [reply]
The API of etherscan is just for Kovan, Rinkeby and Mainnet, if you deploy to a fork or to development, be sure the verify is set up as False.

**Reply by formulaRoot** [reply]
Got it! Thanks @cromewar Appreciate it!
